Userland/Utilities/paste.cpp

```diff
@@ -7,39 +7,34 @@
 #include <AK/Format.h>
 #include <AK/String.h>
 #include <LibCore/ArgsParser.h>
+#include <LibCore/System.h>
 #include <LibGUI/Application.h>
 #include <LibGUI/Clipboard.h>
+#include <LibMain/Main.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/wait.h>
 #include <unistd.h>
 
-static void spawn_command(const Vector<const char*>& command, const ByteBuffer& data, const char* state)
+static ErrorOr<void> spawn_command(const Vector<const char*>& command, const ByteBuffer& data, const char* state)
 {
-    int pipefd[2];
-    if (pipe(pipefd) < 0) {
-        perror("pipe");
-        return;
-    }
+    auto pipefd = TRY(Core::System::pipe());
 
-    pid_t pid = fork();
-    if (pid < 0) {
-        perror("fork");
-        return;
-    } else if (pid == 0) {
+    pid_t pid = TRY(Core::System::fork());
+    if (pid == 0) {
         // We're the child.
-        dup2(pipefd[0], 0);
-        close(pipefd[0]);
-        close(pipefd[1]);
+        TRY(Core::System::dup2(pipefd[0], 0));
```

Comment:
This has some weird semantics, the forked process will end up back in `main`, it's generally considered _weird_ for a forked process to leave its parent's last frame unless it either `exec`'s or `exit`'s.

Reply:
Not entirely sure what to improve there, missing skills.

Reply:
We could use `MUST` here for now so that we'll still exit

Reply:
Oh I see what's the problem now. `MUST` would do.